File: src/main/java/test/com/teambr/bookshelf/common/blocks/TestBlock.java
package test.com.teambr.bookshelf.common.blocks;

import com.teambr.bookshelf.Bookshelf;
import com.teambr.bookshelf.client.itemtooltip.IItemTooltip;
import com.teambr.bookshelf.common.blocks.BaseBlock;
import net.minecraft.block.material.Material;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.world.World;
import test.com.teambr.bookshelf.common.tiles.TileTestBlock;

import java.util.ArrayList;
import java.util.List;

public class TestBlock extends BaseBlock implements IItemTooltip {

    public TestBlock(Material mat, String name, Class<? extends TileEntity> tile) {
        super(mat, name, tile);

        this.setBlockName(name);
    }

    @Override
    public boolean onBlockActivated(World world, int x, int y, int z, EntityPlayer player, int par6, float par7, float par8, float par9) {
        TileTestBlock tile = (TileTestBlock) world.getTileEntity(x, y, z);
        if (tile != null) {
            player.openGui(Bookshelf.instance, 0, world, x, y, z);

            return true;
        }
        return true;
    }

    @Override
    public List<String> returnTooltip() {
        List<String> tip = new ArrayList<>();
        tip.add("This is a test");
        tip.add("Line 2");
        return tip;
    }
}

File: src/main/java/test/com/teambr/bookshelf/common/blocks/TestBlock.scala
package test.com.teambr.bookshelf.common.blocks

import com.teambr.bookshelf.collections.BlockTextures
import com.teambr.bookshelf.common.blocks.BlockBase
import net.minecraft.block.material.Material
import net.minecraft.client.renderer.texture.IIconRegister
import test.com.teambr.bookshelf.common.tiles.TileTestBlock

/**
 * This file was created for the Bookshelf
 * API. The source is available at: 
 * https://github.com/TeamBR-Modding/Bookshelf
 *
 * Bookshelf if licensed under the is licensed under a
 * Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License:
 * http://creativecommons.org/licenses/by-nc-sa/4.0/
 *
 * @author Paul Davis <pauljoda>
 * @since August 02, 2015
 */
class TestBlock extends BlockBase(Material.rock, "Bookshelf:testing", classOf[TileTestBlock]) {
    override def generateDefaultTextures(iconRegistry : IIconRegister) = {
        this.blockIcon = iconRegistry.registerIcon("minecraft:stone")
        textures = new BlockTextures(iconRegistry, "minecraft:stone")
    }
}
